packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -412,6 +443,9 @@ abstract class ImageStreamCompleter with Diagnosticable {
         callback();
       }
       _onLastListenerRemovedCallbacks.clear();
+      if (!keepAlive && _currentImage?.autoDispose == true) {
+        _currentImage!.image.dispose();
```

Comment:
Also, the ownership model of the image seems a little wonky. Is ImageStreamCompleter really authorized to call dispose on it? Not clear to me...

Reply:
In fact, the Image widget appears to remove its listener when the surrounding TickerMode is disabled (which presumably would dispose the image here), but the widget continues to hold on to the image object and continues to render it on screen. Most likely that will crash badly since the image is disposed? (If we don't have it, we should add a test for an Image widget inside a disabled TickerMode)

Reply:
Hmm.

We could have the image widget tell the completer to keep the image alive even if listeners go to zero while the state is still around.

I agree this is not ideal.But we can't have the widget state own this, and we really need something to tell us when it's done with the image.

Reply:
That sounds like a pretty cumbersome API and AFAIK it's not really a pattern we use elsewhere in the framework.